R/Rules-methods.R

```diff
@@ -1001,6 +1001,295 @@ setMethod(
   }
 )
 
+## NextBestProbMTDLTE ----
+
+#' @describeIn nextBest find the next best dose based with the highest
+#'  probability of having a toxicity rate less or equal to the target toxicity
+#'  level.
+#'
+#' @aliases nextBest-NextBestProbMTDLTE
+#'
+#' @export
+#' @example examples/Rules-method-nextBest-NextBestProbMTDLTE.R
+#'
+setMethod(
+  f = "nextBest",
+  signature = signature(
+    nextBest = "NextBestProbMTDLTE",
+    doselimit = "numeric",
+    samples = "Samples",
+    model = "GeneralModel",
+    data = "Data"
+  ),
+  definition = function(nextBest, doselimit, samples, model, data, ...) {
+
+    # Matrix with samples from the dose-tox curve at the dose grid points.
+    prob_samples <- sapply(data@doseGrid, prob, model = model, samples = samples)
+
+    # Determine the maximum dose level with a toxicity probability below or
+    # equal to the target and calculate how often a dose is selected as MTD
+    # across iterations.
+    # The first element of the vector is the relative frequency that no
+    # dose in the grid is below or equal to the target, the
+    # second element that the 1st dose is the MTD, etc.
```

Comment:
the second element could be an active dose if no placebo is included OR a placebo if placebo is included in the data structure. We might need to use a more generic wording perhaps  or sped a few more words to cover this or clarify this..

Reply:
change:
second element that the 1st dose is the MTD
to
second element that the 1st dose of **the grid** is the MTD

Reply:
text updated